Bottom: 6205f2116434fbdf281ee0bc630542795302f40d
Top:    7cfdcc20594ecb37f99fbfb42074c5a89894969d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 18:11:17 +0000

AddElementToBuffer(): Let CopyElementLowLevel() create the element to copy into.


---

diff --git a/src/buffer.c b/src/buffer.c
index 44b181c..5835788 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -211,11 +211,7 @@ AddPolygonToBuffer (LayerType *Layer, PolygonType *Polygon)
 static void *
 AddElementToBuffer (ElementType *Element)
 {
-  ElementType *element;
-
-  element = GetElementMemory (Dest);
-  CopyElementLowLevel (Dest, element, Element, false, 0, 0, FOUNDFLAG | ExtraFlag);
-  return (element);
+  return CopyElementLowLevel (Dest, NULL, Element, false, 0, 0, FOUNDFLAG | ExtraFlag);
 }
 
 /* ---------------------------------------------------------------------------
